pkg/nsx/services/inventory/compare.go

```diff
@@ -0,0 +1,90 @@
+}
+
+func compareContainerProject(pre interface{}, cur interface{}, property *map[string]interface{}) {
+	updateProperties := *property
+	if pre == nil {
+		updateProperties["display_name"] = cur.(containerinventory.ContainerProject).DisplayName
+		updateProperties["container_cluster_id"] = cur.(containerinventory.ContainerProject).ContainerClusterId
+	}
+	if pre == nil || !reflect.DeepEqual(pre.(containerinventory.ContainerProject).Tags, cur.(containerinventory.ContainerProject).Tags) {
+		updateProperties["tags"] = cur.(containerinventory.ContainerProject).Tags
+	}
+
+}
+
+func compareContainerApplicationInstance(pre interface{}, cur interface{}, property *map[string]interface{}) {
+	updateProperties := *property
+	if pre == nil {
+		updateProperties["display_name"] = cur.(containerinventory.ContainerApplicationInstance).DisplayName
+		updateProperties["container_cluster_id"] = cur.(containerinventory.ContainerApplicationInstance).ContainerClusterId
+		updateProperties["container_project_id"] = cur.(containerinventory.ContainerApplicationInstance).ContainerProjectId
+	}
+	if pre == nil || !reflect.DeepEqual(pre.(containerinventory.ContainerApplicationInstance).Tags, cur.(containerinventory.ContainerApplicationInstance).Tags) {
+		updateProperties["tags"] = cur.(containerinventory.ContainerApplicationInstance).Tags
+	}
+	if pre == nil || pre.(containerinventory.ContainerApplicationInstance).ClusterNodeId != cur.(containerinventory.ContainerApplicationInstance).ClusterNodeId {
+		updateProperties["cluster_node_id"] = cur.(containerinventory.ContainerApplicationInstance).ClusterNodeId
+	}
+	if pre == nil || !reflect.DeepEqual(pre.(containerinventory.ContainerApplicationInstance).ContainerApplicationIds, cur.(containerinventory.ContainerApplicationInstance).ContainerApplicationIds) {
+		updateProperties["container_application_ids"] = cur.(containerinventory.ContainerApplicationInstance).ContainerApplicationIds
+	}
+	if pre == nil || pre.(containerinventory.ContainerApplicationInstance).Status != cur.(containerinventory.ContainerApplicationInstance).Status {
+		updateProperties["status"] = cur.(containerinventory.ContainerApplicationInstance).Status
+	}
+	if pre == nil && len(cur.(containerinventory.ContainerApplicationInstance).OriginProperties) == 1 {
+		updateProperties["origin_properties"] = cur.(containerinventory.ContainerApplicationInstance).OriginProperties
+	} else if pre != nil && !reflect.DeepEqual(pre.(containerinventory.ContainerApplicationInstance).OriginProperties, cur.(containerinventory.ContainerApplicationInstance).OriginProperties) {
+		if isIPChanged(pre.(containerinventory.ContainerApplicationInstance), cur.(containerinventory.ContainerApplicationInstance)) {
+			updateProperties["origin_properties"] = cur.(containerinventory.ContainerApplicationInstance).OriginProperties
+		}
+	}
+}
+
+func isIPChanged(pre containerinventory.ContainerApplicationInstance, cur containerinventory.ContainerApplicationInstance) bool {
```

Comment:
rename it to isIPChangedContainerApplicationInstance?

Reply:
the context and the parameter could give enough info. and it's used internal, long func name is not necessary. Func name and the parameter have duplicate info